test/DurableTask.SqlServer.Tests/Integration/DatabaseManagement.cs

```diff
@@ -127,7 +128,61 @@ public async Task CanCreateAndDropSchema(bool isDatabaseMissing)
             // We know that all objects were successfully removed if the "dt" no longer exists.
             Assert.DoesNotContain("dt", testDb.GetSchemas());
         }
+        
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task CanCreateAndDropSchemaWithCustomSchemaName(bool isDatabaseMissing)
+        {
+            using TestDatabase testDb = this.CreateTestDb(!isDatabaseMissing);
+            string schemaName = "testSchemaName";
+            IOrchestrationService service = this.CreateServiceWithTestDb(testDb,schemaName);
 
+            await service.CreateAsync(recreateInstanceStore: true);
+            
+            LogAssert.NoWarningsOrErrors(this.logProvider);
+            LogAssert
+                .For(this.logProvider)
+                .Expect(
+                    LogAssert.CheckedDatabase())
+                .ExpectIf(
+                    isDatabaseMissing,
+                    LogAssert.CommandCompleted($"CREATE DATABASE [{testDb.Name}]"),
+                    LogAssert.CreatedDatabase(testDb.Name))
+                .Expect(
+                    LogAssert.AcquiredAppLock(),
+                    LogAssert.ExecutedSqlScript("drop-schema.sql"),
+                    LogAssert.ExecutedSqlScript("schema-1.0.0.sql"),
+                    LogAssert.ExecutedSqlScript("logic.sql"),
+                    LogAssert.ExecutedSqlScript("permissions.sql"),
+                    LogAssert.SprocCompleted($"{schemaName}._UpdateVersion"))
+                .EndOfLog();
+            
+            ValidateDatabaseSchema(testDb, schemaName);
+            
+            this.logProvider.Clear();
+            await service.CreateIfNotExistsAsync();
+            ValidateDatabaseSchema(testDb, schemaName);
+            
+            LogAssert.NoWarningsOrErrors(this.logProvider);
+            LogAssert.Sequence(
+                this.logProvider,
+                LogAssert.CheckedDatabase(),
+                LogAssert.AcquiredAppLock(),
+                LogAssert.SprocCompleted($"{schemaName}._GetVersions"));
+            
+            this.logProvider.Clear();
+            await service.DeleteAsync();
+            LogAssert.NoWarningsOrErrors(this.logProvider);
+            LogAssert.Sequence(
+                this.logProvider,
+                LogAssert.AcquiredAppLock(),
+                LogAssert.ExecutedSqlScript("drop-schema.sql"));
+            
+            Assert.DoesNotContain($"{schemaName}", testDb.GetSchemas());
+            Assert.DoesNotContain("dt", testDb.GetSchemas());
+
+        }
```

Comment:
Can we add another test that adds multiple schemas to the same database? For example, the database is initially created with the "dt" schema and then we later set up an orchestration service using the "testSchemaName" schema. The end result should be two sets of tables. Even better if we can also verify that executing the drop scripts only impacts the target schema when multiple schemas are present.

Reply:
Added a new test to verify that dropping one schema won't affect the other one and two sets of tables are created with the right names. Also added a few parameters to Get* functions to interact with the logger correctly.